token/program-2022/src/extension/confidential_transfer/instruction.rs

```diff
@@ -419,8 +425,35 @@ pub enum ConfidentialTransferInstruction {
     HarvestWithheldTokensToMint,
 }
 
+/// Data expected by `ConfidentialTransferInstruction::InitializeMint`
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable)]
+#[repr(C)]
+pub struct InitializeMintData {
+    /// Authority to modify the `ConfidentialTransferMint` configuration and to approve new
+    /// accounts.
+    pub authority: Pubkey,
+    /// Determines if newly configured accounts must be approved by the `authority` before they may
+    /// be used by the user.
+    pub auto_approve_new_accounts: PodBool,
+    /// New authority to decode any transfer amount in a confidential transfer.
+    pub auditor_encryption_pubkey: EncryptionPubkey,
+    /// Authority to withdraw withheld fees that are associated with accounts.
+    pub withdraw_withheld_authority_encryption_pubkey: EncryptionPubkey,
+}
+
+/// Data expected by `ConfidentialTransferInstruction::UpdateMint`
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable)]
+#[repr(C)]
+pub struct UpdateMintData {
+    /// Determines if newly configured accounts must be approved by the `authority` before they may
+    /// be used by the user.
+    pub auto_approve_new_accounts: PodBool,
+    /// New authority to decode any transfer amount in a confidential transfer.
+    pub auditor_encryption_pubkey: EncryptionPubkey,
+}
```

Comment:
How can we set `withdraw_withheld_authority_encryption_pubkey`? Is the idea that it's only possible when there are no withheld fees, which can never be guaranteed?

Reply:
Correct, the `withdraw_withheld_authority_encryption_pubkey` can only safely be updated when there are no (confidential extension) accounts with non-zero withheld fees. I created issue #3942 discussing this issue.